qwen_agent/llm/openvino.py

```diff
@@ -0,0 +1,154 @@
+from typing import Dict, Iterator, List, Optional
+from threading import Thread
+
+from optimum.intel.openvino import OVModelForCausalLM
+from transformers import (AutoTokenizer, AutoConfig,
+                          TextIteratorStreamer, StoppingCriteriaList, StoppingCriteria)
+
+from qwen_agent.llm.base import register_llm
+from qwen_agent.llm.schema import ASSISTANT, DEFAULT_SYSTEM_MESSAGE, SYSTEM, USER, Message
+from qwen_agent.llm.text_base import BaseTextChatModel
+from qwen_agent.log import logger
+
+
+from transformers.generation.stopping_criteria import StoppingCriteriaList, StoppingCriteria
+
+
+class StopSequenceCriteria(StoppingCriteria):
+    """
+    This class can be used to stop generation whenever a sequence of tokens is encountered.
+
+    Args:
+        stop_sequences (`str` or `List[str]`):
+            The sequence (or list of sequences) on which to stop execution.
+        tokenizer:
+            The tokenizer used to decode the model outputs.
+    """
+
+    def __init__(self, stop_sequences, tokenizer):
+        if isinstance(stop_sequences, str):
+            stop_sequences = [stop_sequences]
+        self.stop_sequences = stop_sequences
+        self.tokenizer = tokenizer
+
+    def __call__(self, input_ids, scores, **kwargs) -> bool:
+        decoded_output = self.tokenizer.decode(input_ids.tolist()[0])
+        return any(decoded_output.endswith(stop_sequence) for stop_sequence in self.stop_sequences)
+
+
+@register_llm('openvino')
+class OpenVINO(BaseTextChatModel):
+    """OpenVINO Pipeline API.
+
+    To use, you should have the ``optimum[openvino,nncf]`` python package installed.
```

Comment:
nncf dependencies included into optimum[openvino] extras now
```suggestion
    To use, you should have the ``optimum[openvino]`` python package installed.
```